src/main/java/io/reactivex/internal/operators/flowable/FlowableCombineLatest.java

```diff
@@ -73,6 +76,7 @@ public void subscribeActual(Subscriber<? super R> s) {
         Publisher<? extends T>[] a = array;
         int n;
         if (a == null) {
+            assert iterable != null; //either array or iterable are initialized with non null values
```

Comment:
Asserts are off by default but they increase the bytecode size; besides we don't use `assert` it anyway.